webrtc/rtcpeerconnection/setLocalDescription.html

```diff
@@ -0,0 +1,30 @@
+<!doctype html>
+<html>
+<head>
+  <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
+  <title>RTCPeerConnection setLocalDescription tests</title>
+</head>
+<body>
+  <!-- These files are in place when executing on W3C. -->
+  <script src="/resources/testharness.js"></script>
+  <script src="/resources/testharnessreport.js"></script>
+  <script type="text/javascript">
+  // tests that the icegatheringstatechange event is called after calling setLocalDescription.
+  async_test(function(test) {
+    var pc = new RTCPeerConnection(null);
+
+    pc.onicegatheringstatechange = test.step_func(function(event) {
+      if (pc.iceGatheringState === 'complete') {
+        test.done();
+      }
+    });
+    pc.createOffer({offerToReceiveAudio: 1})
+    .then(offer => pc.setLocalDescription(offer))
+    .catch(test.step_func(function(e) {
+      assert_unreached('Error ' + e.name + ': ' + e.message);
+    }));
+  }, 'Triggers onicegatheringstatechange with "complete" after creating an offer and calling setLocalDescription.');
```

Comment:
I'd prefer to call this "Tests that the ICE gathering state ends up as "completed" after applying a local offer".